pneumonia_functions.py

```diff
@@ -0,0 +1,280 @@
+    names_file = os.path.join(train_data_dir, "obj.names")
+    with open(names_file, 'w') as names:
+        for obj_name in list_names:
+            line = "{}\n".format(obj_name)
+            names.write(line)
+
+
+def yolo_data_file(train_data_dir, backup, class_nbr):
+    '''Generate the file with paths for yolo v3 training
+    The file will be save under 'obj.data' in data directory'''
+    data_file = os.path.join(train_data_dir, "obj.data")
+    with open(data_file, 'w') as data:
+        line = f"classes = {class_nbr}\n"\
+        f"train = {train_data_dir + 'train.txt'}\n"\
+        f"valid = {train_data_dir + 'val.txt'}\n"\
+        f"names = {train_data_dir + 'obj.names'}\n"\
+        f"backup = {backup}"
+        data.write(line)
+
+
+def dcm_to_array(image_path):
+    '''Tranform dicom image format to a numpy array'''
+    dcm_im = pydicom.read_file(image_path + ".dcm").pixel_array
+    return np.stack([dcm_im]*3, -1)
+
+
+def yolo_jpg_file(dataset, origin_folder, target_folder):
+    '''Copy the choosen images in the right directory under jpg format'''
+    for image_name in dataset.iloc[:, 0].unique():
+        image = dcm_to_array(os.path.join(origin_folder, image_name))
+        cv2.imwrite(os.path.join(target_folder, image_name + ".jpg"), image)
+
+
+def yolo_label_generation(dataset, target_folder, image_size):
+    '''Generate label in the shape of yolo_v3 learning CNN:
+    <object-class> <x_center> <y_center> <width> <height>
+    relative value is required by yolo_v3 algorithm.
+    one txt file per image is also required by yolo_v3 algorithm
+    As mentionned in darknet repo, to improve results we need to add images without objects
+    All created files will be saved in the same directory which contains jpg files'''
+    for name, groupe in dataset.groupby("patientId"):
+        label_file = os.path.join(target_folder, name + ".txt")
+        with open(label_file, "w+") as file:
+            for x, y, w, h, cl in groupe.iloc[:, 1:].values:
+                if cl:
+                    rel_w = w / image_size
+                    rel_h = h / image_size
+                    rel_x_center = x / image_size + rel_w / 2
+                    rel_y_center = y / image_size + rel_h / 2
+                    line = f"{int(cl - 1)} "\
+                    f"{rel_x_center} "\
+                    f"{rel_y_center} "\
+                    f"{rel_w} "\
+                    f"{rel_h}\n"
+                    file.write(line)
+
+
+def yolo_image_path_file(dataset, target_folder, train_images_dir, file_name):
+    '''Generate a 'txt' file with the path and the name of each image'''
+    txt_file = os.path.join(target_folder, file_name)
+    with open(txt_file, "w+") as file:
+        for image_name in dataset.iloc[:, 0].unique():
+            line = "{}\n".format(os.path.join(train_images_dir, image_name + ".jpg"))
+            file.write(line)
+
+
+def yolo_pre_trained_weights(link, path):
```

Comment:
misnamed function: this is not an action name. Also it's quite useless: it's a oneliner who is quite clear...